addons/arsenal/functions/fnc_onSelChangedLeft.sqf

```diff
@@ -18,429 +19,631 @@ params ["_control", "_curSel"];
 
 if (_curSel < 0) exitwith {};
 
-private _ctrlIDC = ctrlIDC _control;
 private _display = ctrlParent _control;
-private _item = [_control lnbData [_curSel, 0], _control lbData _curSel] select !(ctrlType _control == 102);
+private _item = [_control lbData _curSel, _control lnbData [_curSel, 0]] select (ctrlType _control == CT_LISTNBOX);
 
-private _weaponDefaultRightPanel = _display displayCtrl IDC_buttonOptic;
-private _selectCorrectPanelWeapon = [_weaponDefaultRightPanel, _display displayCtrl GVAR(currentRightPanel)] select (!isNil QGVAR(currentRightPanel) && {GVAR(currentRightPanel) in [RIGHT_PANEL_ACC_IDCS, IDC_buttonCurrentMag, IDC_buttonCurrentMag2]});
-
-private _containerDefaultRightPanel = _display displayCtrl IDC_buttonMisc;
-private _selectCorrectPanelContainer = [_containerDefaultRightPanel, _display displayCtrl GVAR(currentRightPanel)] select (!isNil QGVAR(currentRightPanel) && {GVAR(currentRightPanel) in [RIGHT_PANEL_ITEMS_IDCS]});
+// When having chosen a new category, see if the current right panel can be kept open, otherwise take default
+private _selectCorrectPanelWeapon = [_display displayCtrl IDC_buttonOptic, _display displayCtrl GVAR(currentRightPanel)] select (!isNil QGVAR(currentRightPanel) && {GVAR(currentRightPanel) in [RIGHT_PANEL_ACC_IDCS, IDC_buttonCurrentMag, IDC_buttonCurrentMag2]});
+private _selectCorrectPanelContainer = [_display displayCtrl IDC_buttonMisc, _display displayCtrl GVAR(currentRightPanel)] select (!isNil QGVAR(currentRightPanel) && {GVAR(currentRightPanel) in [RIGHT_PANEL_ITEMS_IDCS]});
 
+// Remove all magazines from the previous weapon that aren't compatible with the new weapon
 private _fnc_clearPreviousWepMags = {
-    private _compatibleMagsBaseWeapon = [_baseWeapon, true] call CBA_fnc_compatibleMagazines;
+    private _compatibleMagsBaseWeapon = compatibleMagazines _baseWeapon;
```

Comment:
Can be reverted if wished.

Reply:
Decision has been taken to move forward with engine commands instead of CBA. If bugs are found with engine commands, ACE will revert back to the CBA functions.